CMakeLists.txt

```diff
@@ -54,7 +56,7 @@ OPTION(BUILD_DOCS "Set to ON to build documentation" OFF)
 #   Find Dependencies
 # ----------------------------------------------------------------------------
 
-IF(BUILD_EXAMPLES)
+IF(USE_OPENCV)
```

Comment:
@beru
In this PR CMake requires opencv even if we don't compile the mnist example (This is why AppVeyor failed to build). We need opencv only when BUILD_EXAMPLS && USE_OPENCV. Could you fix it?